docs/adr/0002-database-corruption.md

```diff
@@ -0,0 +1,132 @@
+# Handling Database Corruption
+
+* Status: accepted
+* Date: 2021-06-08
+
+## Context and Problem Statement
+
+Some of our users have corrupt SQLite databases and this makes the related
+component unusable.  The best way to deal with corrupt databases is to simply
+delete the database and start fresh (see #2628).  However, it's not so easy
+to know if a database operation error is from a corrupt database or from bad
+code.  We want a system that identifies and deals with database corruption but
+avoids deleting data based on logic errors.  That system needs to deal with 3
+related questions:
+
+  * A) When do we identify corrupted databases?
+  * B) How do we identify corrupted databases?
+  * C) What do we do when we identify corrupted databases?
+
+## Decision Drivers
+
+* Deleting valid user data should be avoided at almost any cost
+* Keeping a corrupted database around is almost as bad.  It currently prevents
+  the component from working at all.
+* We don't currently have a good way to distinguish between the two error
+  cases, but this can be improved by reviewing telemetry and sentry data.
+
+## Considered Options
+
+* A) When do we identify corrupted databases?
+  * At migration time only
+  * Whenever we perform a database operation
+* B) How do we identify corrupted databases?
+  * Assume all errors during migration are from corrupt databases
+  * Compare error values against known corruption errors for all operations
+  * Compare error values against known corruption errors when opening the database
+* C) What do we do when we identify corrupted databases?
+  * Delete the database file and recreate the database
+  * Move the database file and recreate the database
+  * Have the component fail
+
+## Decision Outcome
+
+* A: Whenever we perform a database operation
+
+This was chosen because it can be a long time between migrations and we want
+to de able to fix corrupt database during that period.
+
+* B: Compare error values against known corruption errors when opening the database
```

Comment:
Eg, A and B are contradictory here - A) says we will identify whenever we perform a database operation and B) says only when opening